pkg/connector/user.go

```diff
@@ -151,7 +151,7 @@ func (o *userResourceType) List(ctx context.Context, parentID *v2.ResourceId, pt
 		u, res, err := o.client.Users.GetByID(ctx, user.GetID())
 		if err != nil {
 			// This undocumented API can return 404 for some users. If this fails it means we won't get some of their details like email
-			if res.StatusCode != http.StatusNotFound {
+			if res == nil || res.StatusCode != http.StatusNotFound {
```

Comment:
I wouldn't trust anything at the left of an `err` tbh 😁

Reply:
Yeah but we want to check the status code. 🤷

Reply:
We keep trucking even though we don't have a user, and use a presumably nil value later?  Does this move the panic down the stack?

Reply:
We set u to user on line 158.

Reply:
You can call methods on nil pointer receivers.  I didn't see reseting the user on 158, so +1.